Cakephp datafields can be 'modified' or 'updated'.

nops opened the pull request:
# Model.php

$dateFields = array('modified', 'updated');

nops committed:
Cakephp datafields can be 'modified' or 'updated'.
#Model.php
$dateFields = array('modified', 'updated');

AD7six commented:
this change is going to lead to confusing results (and there's no test for it). if 'updated' is in the data passed to save, it'll use that field. if it's not, it will use modified.

Personally, I'd rather datasources (and cake) did not do this, and users or a behavior (or the datastore itself where appropraite) took care of populating updated/modified/changed/tweaked fields as the develop sees fit.

nops commented:
it has absolutely no change if you don't use the updated field. Cakephp, for convention, change both fields "auto-magic" on save as you can see in http://book.cakephp.org/view/1015/created-and-modified. If you save with 'mongoNoSetOperator' and updated field, it breaks.

ichikaway commented:
sorry late reply. 
Thank you for the patch.
I merged it.

> AD7six

In this case, cake Model::save automatically set datetime to modified or updated field if there is the field in save data.
This datasource sets $set for the field, because combination with other operators as follows.
$savedata = array(
  '_id' => xxx1, 
 '$inc' => array('count' => 1),
 'updated' => '2011/8/1',
);

This savedata will be converted as follows in the update() method in this datasource.
$savedata = array(
  '_id' => xxx1, 
 '$inc' => array('count' => 1),
 '$set' => array('updated' => '2011/8/1'),
);
